pkg/cri/server/container_create_linux.go

```diff
@@ -601,5 +601,5 @@ func generateUserString(username string, uid, gid *runtime.Int64Value) (string,
 
 // snapshotterOpts returns any Linux specific snapshotter options for the rootfs snapshot
 func snapshotterOpts(snapshotterName string, config *runtime.ContainerConfig) []snapshots.Opt {
-	return []snapshots.Opt{}
+	return []snapshots.Opt{snapshots.WithLabels(snapshots.FilterInheritedLabels(config.Annotations))}
```

Comment:
should inherit only what pass the size validation https://github.com/containerd/containerd/blob/main/labels/validate.go

Reply:
oh yes, I agree with you.

Reply:
Should I verify the labels validated here or maybe check the validation of all snapshot labels in function FilterInheritedLabels()?